egs/wsj/s5/utils/make_lexicon_fst.pl

```diff
@@ -148,18 +135,16 @@ sub is_sil {
         $word_or_eps = "<eps>";
         $pron_cost_string = ""; $pron_cost = 0.0; # so we only print it the 1st time.
         $s = $ns;
+      } elsif ($p ne $silphone) {
```

Comment:
could you please make this
!defined($silphone) || $p ne $silphone
and re-test?  Also test without specifying silprob and silphone on the command line.
I am concerned otherwise we may get spurious warnings about undefined comparisons
Dan